notification-sdk/src/main/java/org/sunbird/notification/sms/providerimpl/Msg91SmsProviderImpl.java

```diff
@@ -392,26 +404,143 @@ public boolean bulkSms(List<String> phoneNumber, String smsText) {
-}
+	public boolean verifyOtp(OTPRequest request) {
+		boolean response = false;
+		try {
+			HttpResponse<String> resendResponse = Unirest
+					.get(OTP_BASE_URL + "verifyRequestOTP.php?authkey="
+							+ Util.readValue(NotificationConstant.SUNBIRD_MSG_91_AUTH) + NotificationConstant.Ampersand
+							+ NotificationConstant.MOBILE + NotificationConstant.EQUAL + request.getCountryCode()
+							+ request.getPhone() + NotificationConstant.Ampersand + NotificationConstant.OTP
+							+ NotificationConstant.EQUAL + request.getOtp())
+					.header("content-type", "application/x-www-form-urlencoded").asString();
+
+			if (resendResponse != null) {
+				if (resendResponse.getStatus() == NotificationConstant.SUCCESS_CODE) {
+					logger.info("OTP verify response data " + resendResponse.getBody());
+					MessageResponse messageResponse = convertMsg91Response(resendResponse.getBody());
+					if (NotificationConstant.SUCCESS.equalsIgnoreCase(messageResponse.getType())) {
+						logger.info("OTP verify succssfully with response data " + resendResponse.getBody());
+						response = true;
+					}
+				} else {
+					logger.info("OTP verification failed with code and response data " + resendResponse.getStatus()
+							+ " -" + resendResponse.getBody());
+				}
+
+			} else {
+				logger.info("OTP verification failed .");
+			}
+
+		} catch (Exception e) {
+			logger.error("Msg91SmsProviderImpl:sendOtp  exception occured during otp verification :" + e.getMessage());
+		}
+		return response;
+	}
+
+
+	private String createOtpReqData(OTPRequest request) {
+		StringBuilder builder = new StringBuilder();
+		if (StringUtils.isNotBlank(request.getOtp())) {
+			builder.append(NotificationConstant.Ampersand + NotificationConstant.OTP + NotificationConstant.EQUAL
+					+ request.getOtp());
+		} else {
+			builder.append(NotificationConstant.Ampersand + NotificationConstant.SENDER + NotificationConstant.EQUAL
+					+ Util.readValue(NotificationConstant.SUNBIR_MSG_DEFAULT_SENDER));
+		}
+		try {
+			builder.append(NotificationConstant.Ampersand + NotificationConstant.MESSAGE + NotificationConstant.EQUAL
+					+ URLEncoder.encode(request.getMessage(), "UTF-8") + NotificationConstant.Ampersand
+					+ NotificationConstant.MOBILES + NotificationConstant.EQUAL + request.getCountryCode()
+					+ request.getPhone());
+		} catch (UnsupportedEncodingException e) {
+			e.printStackTrace();
+		}
+		builder.append(NotificationConstant.Ampersand + NotificationConstant.OTP_LENGTH + NotificationConstant.EQUAL
+				+ request.getOtpLength() + NotificationConstant.Ampersand + NotificationConstant.OTP_EXPIRY
+				+ NotificationConstant.EQUAL + request.getExpiryTimeInMinute());
+		return builder.toString();
+
+	}
+	
+	
+	private MessageResponse convertMsg91Response(String response) {
+		MessageResponse messageResponse = new MessageResponse();
+		try {
+			messageResponse = mapper.readValue(response, MessageResponse.class);
+		} catch (JsonParseException e) {
+			e.printStackTrace();
```

Comment:
Sir we need to add Logger.ERROR instead of e.printStackTrace()

Reply:
done